os/pm/pm_idle.c

```diff
@@ -84,7 +94,35 @@ void pm_idle(void)
 		/* Decide, which power saving level can be obtained */
 		newstate = pm_checkstate();
 		/* Perform state-dependent logic here */
-		pmvdbg("newstate= %d\n", newstate);
+		/* For SMP case, we need to check secondary core status
+	     * If secondary core status is not in idle thread, abort
+	     * the sleep and check again on next cycle
+		 */
+#ifdef CONFIG_SMP
+		if (newstate == PM_SLEEP) {
+			for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
+				/* If the CPU is just back from sleep, abort the sleep */
+				if (up_get_secondary_cpu_state(cpu) == CPU_WAKE_FROM_SLEEP) {
```

Comment:
Can we change this api name as up_get_cpu_state and up_set_cpu_state? 
As these apis are used for all cpus in the system, the name secondary_cpu seems wrong.

Reply:
I agree, renamed it accordingly